src/ds/ccf_deprecated.h

```diff
@@ -0,0 +1,5 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#define CCF_DEPRECATED(reason) [[deprecated(reason)]]
```

Comment:
Why do we wrap this? That's standard as of 14, the wrapper isn't shorter.

Reply:
The main reason is that `clang-format` seemed very confused by this attribute. Either the formatter splits it across multiple lines, or we add extra lines with `//clang-format off/on` - the macro keeps this as a single line.

As minor side advantages we can easily grep for exactly these lines, we have a single place to disable them if we ever need to, we can add versioned macros (`CCF_DEPRECATED_1_0_1`) to find the things removed in specific versions.

Reply:
That seems like the kind of thing upgrading to a newer clang-format and/or tweaking the config might solve?

I do like the versioned deprecated, especially if we had an automated check for it (ie. cutting X.Y fails until all deprecated X.Y are removed).